S3Sample.java

```diff
@@ -0,0 +1,242 @@
+             * keep your data closer to your applications or users.
+             */
+            System.out.println("Creating bucket " + bucketName + "\n");
+            s3.createBucket(bucketName);
+
+            /*
+             * List the buckets in your account
+             */
+            System.out.println("Listing buckets");
+            for (Bucket bucket : s3.listBuckets()) {
+                System.out.println(" - " + bucket.getName());
+                Thread.sleep(2000);
+                Thread.sleep(2000);
+                Thread.sleep(2000);
+            }
+            System.out.println();
+
+            /*
+             * Upload an object to your bucket - You can easily upload a file to
+             * S3, or upload directly an InputStream if you know the length of
+             * the data in the stream. You can also specify your own metadata
+             * when uploading to S3, which allows you set a variety of options
+             * like content-type and content-encoding, plus additional metadata
+             * specific to your applications.
+             */
+            System.out.println("Uploading a new object to S3 from a file\n");
+            s3.putObject(new PutObjectRequest(bucketName, key, createSampleFile()));
+
+            /*
+             * Download an object - When you download an object, you get all of
+             * the object's metadata and a stream from which to read the contents.
+             * It's important to read the contents of the stream as quickly as
+             * possibly since the data is streamed directly from Amazon S3 and your
+             * network connection will remain open until you read all the data or
+             * close the input stream.
+             *
+             * GetObjectRequest also supports several other options, including
+             * conditional downloading of objects based on modification times,
+             * ETags, and selectively downloading a range of an object.
+             */
+            System.out.println("Downloading an object");
+            S3Object object = s3.getObject(new GetObjectRequest(bucketName, key));
+            System.out.println("Content-Type: "  + object.getObjectMetadata().getContentType());
+            displayTextInputStream(object.getObjectContent());
+
+            /*
+             * List objects in your bucket by prefix - There are many options for
+             * listing the objects in your bucket.  Keep in mind that buckets with
+             * many objects might truncate their results when listing their objects,
+             * so be sure to check if the returned object listing is truncated, and
+             * use the AmazonS3.listNextBatchOfObjects(...) operation to retrieve
+             * additional results.
+             */
+            System.out.println("Listing objects");
+            ObjectListing objectListing = s3.listObjects(new ListObjectsRequest()
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer

This code might not produce accurate results if the operation returns paginated results instead of all results. Consider adding another call to check for additional results.